private/buf/bufworkspace/workspace_targeting.go

```diff
@@ -557,10 +562,28 @@ func getMappedModuleBucketAndModuleTargeting(
 			),
 		)
 	}
+	docBucket := bufmodule.GetDocStorageReadBucket(ctx, moduleBucket)
+	licenseBucket := bufmodule.GetLicenseStorageReadBucket(moduleBucket)
+	if useTopLevelLicenseDoc {
+		docBucketEmpty, err := isBucketEmpty(ctx, docBucket)
```

Comment:
I think this logic could be moved to the `bufmodule` package, with a func signature of `GetDocStorageReadBucket(ctx, bucket, moduleDir, useTopLevelLicenseDoc)`. Then instead of building an empty bucket and checking for emptiness, we can stat the files in order and return a mapped bucket.

Reply:
Do you know if we need to have two separate methods for the buckets and then combine them? Would be nice to use a single set of Matchers with an OR for the doc or license file.